ckine/figures/figureCommon.py

```diff
@@ -437,3 +438,53 @@ def hill_equation(x, x0, solution=0):
 def residuals(x0, x, y):
     """ Residual function for Hill Equation. """
     return hill_equation(x, x0) - y
+
+
+def expScaleWT(predSTAT2, predSTAT15, expSTAT2, expSTAT15, rep2=False):
+    """Scales data to model predictions. It is assumed here that predictions and data are lined up by concentration"""
+    cellGroups = [['NK'], ['CD8+', 'Naive CD8+', 'Mem CD8+'], ['T-reg', 'Naive Treg', 'Mem Treg'], ['T-helper', 'Naive Th', 'Mem Th']]
+    x0 = np.array([0, 1])
+    iterator, iteratorEnd = 0, 0
+    output2 = np.zeros(expSTAT2.shape)
+    output15 = np.zeros(expSTAT15.shape)
+
+    for cellSet in cellGroups:
+        iteratorEnd += len(cellSet)
+        subExpSTAT2 = np.reshape(expSTAT2[(iterator * 4): (iteratorEnd * 4)], (len(cellSet), predSTAT2.shape[3], predSTAT2.shape[1]))
+        subExpSTAT15 = np.reshape(expSTAT15[(iterator * 4): (iteratorEnd * 4)], (len(cellSet), predSTAT15.shape[3], predSTAT15.shape[1]))
+        subPredSTAT2 = predSTAT2[(iterator): (iteratorEnd), :, :, :]
+        subPredSTAT15 = predSTAT15[(iterator): (iteratorEnd), :, :, :]
+
+        subPredSTAT2 = np.swapaxes(subPredSTAT2, 3, 1)
+        subPredSTAT2 = np.swapaxes(subPredSTAT2, 3, 2)
+        subPredSTAT15 = np.swapaxes(subPredSTAT15, 3, 1)
+        subPredSTAT15 = np.swapaxes(subPredSTAT15, 3, 2)
+
+        if rep2:
+            subExpSTAT2 = np.nan_to_num(subExpSTAT2)
```

Comment:
You're replacing nan's with 0's? Isn't that going to affect the results?

Reply:
Was an oversight - replaced with 0 for 2 sets of 2 data points in the IL15 experiments as it was low concentrations. I've replaced this by copying the third smallest concentration into those slots.

Reply:
I think you're better off just removing values for the conversion calculation when one is nan. You can do ravel, then isnan to identify which indices are nan, then remove those indices from x and y.

Reply:
Gotcha - taken care of.